Reimplemented damage logic for RangedEnemy with short-range attack

PizzaE4ter opened the pull request:
Ranged Enemies are able to deal damage now

Pazl27 commented:
could you add also something like a projectile maybe so that it is better visible. Do the ranged enemies always hit the player or can the player doge. If this is not the case how can i avoid them ?

PizzaE4ter commented:
> could you add also something like a projectile maybe so that it is better visible. Do the ranged enemies always hit the player or can the player doge. If this is not the case how can i avoid them ?

That’s what I was thinking too, but I wasn’t sure if this would be the right time to implement the projectile feature as part of this issue.
We could create a separate issue to handle the projectile functionality specifically.

Pazl27 commented:
> > could you add also something like a projectile maybe so that it is better visible. Do the ranged enemies always hit the player or can the player doge. If this is not the case how can i avoid them ?
> 
> That’s what I was thinking too, but I wasn’t sure if this would be the right time to implement the projectile feature as part of this issue. We could create a separate issue to handle the projectile functionality specifically.

i would do this in this branch. just a visual helper for the player so that it isn't complete understandable where the damage is coming from . 
I also would add a mechanic maybe a random chance that the projectile hits the enemy. so that it isn't impossible to outrun key

PizzaE4ter commented:
> > > could you add also something like a projectile maybe so that it is better visible. Do the ranged enemies always hit the player or can the player doge. If this is not the case how can i avoid them ?
> > 
> > 
> > That’s what I was thinking too, but I wasn’t sure if this would be the right time to implement the projectile feature as part of this issue. We could create a separate issue to handle the projectile functionality specifically.
> 
> i would do this in this branch. just a visual helper for the player so that it isn't complete understandable where the damage is coming from . I also would add a mechanic maybe a random chance that the projectile hits the enemy. so that it isn't impossible to outrun key

# Adjust EnemyBase and RangedEnemy cooldown logic, integrate projectile damage

## Changes
### EnemyBase
- Made `attackCooldown` and `timeUntilAttack` `protected virtual` for better extensibility.
- Simplified cooldown logic for reuse in derived classes.

### RangedEnemy
- Overrode `attackCooldown` and `timeUntilAttack` for custom cooldown behavior.
- Passed `damage` to `EnemyProjectile` during initialization.

### EnemyProjectile
- Removed `[Export]` for `Damage`, as it is now dynamically set by the enemy.
- Updated `Initialize` to accept direction and damage.
- Adjusted collision layers and masks to ensure projectiles can hit the player.

## Testing
- Verified cooldown behavior for `RangedEnemy`.
- Ensured projectiles inherit damage from the enemy and apply it correctly to the player.
- Confirmed that projectiles only collide with the player and behave as expected upon collision.

PizzaE4ter commented:
> Hier muss auch noch der master mit multiplayer noch gemerged werden.
> 
> Dazu fliegen bei mir die Pfeile queer könntest du anpassen das die Pfeile gerade fliegen und ein anderes Projektil verwenden. Denn Gegner sollten nicht dieselben projektile wie Spieler benutzen ansonsten weiß man im multiplyer nicht welchen proektilen man ausweichen muss und welchen nicht. Dazu währen knallige farben für gefährliche Projektile gut das die spieler es einfacher haben sie zu lokalisieren.

Bzgl den Pfeilen kann man ein neuen Issue anlegen, die Spielerpfeile habe ich jetzt mal provisorisch angelegt. Was meinst du, dass die Pfeile quer fliegen? Die Feinde schießen die Pfeile immer in Richtung des Spielers

NeXtGhost16 commented:
die pfeile fliegen in richtun g des spielers aber bei mir zeigt die spitze des pfeils nicht auf den spieler sondern zur seite

PizzaE4ter commented:
> die pfeile fliegen in richtun g des spielers aber bei mir zeigt die spitze des pfeils nicht auf den spieler sondern zur seite

das ist momentan so gewollt
